## CMP262-2622 Final Project part 2 Exploratory Analysis
## and part 3 Data Cleaning
### By Dana Tryon
This project will analyze data collected by the New York City Health Department about COVID-19 statistics from 2/29/20 to 4/4/24.  I found this data set on the data.gov website and the URL as follows https://data.cityofnewyork.us/api/views/rc75-m7u3/rows.csv?accessType=DOWNLOAD  

In [1]:
import pandas as pd
df = pd.read_csv(r'C:\Users\Danat\OneDrive\Documents\CMP-262\FinalProject\NYC-COVID-19.csv') 
df.shape  #gives the dimensions of the table in rows by columns

(1501, 55)

The data is organized as a table with 55 columns and 1501 rows.  The first column is the primary key which is the date.  The rest of the columns
are case counts, probable case counts, hospitalization counts, and death counts.  These are city wide totals and are repeated for each of the 5
bouroughs separately.

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1501 entries, 0 to 1500
Data columns (total 55 columns):
 #   Column                           Non-Null Count  Dtype 
---  ------                           --------------  ----- 
 0   date_of_interest                 1501 non-null   object
 1   CASE_COUNT                       1501 non-null   int64 
 2   PROBABLE_CASE_COUNT              1501 non-null   int64 
 3   HOSPITALIZED_COUNT               1501 non-null   int64 
 4   DEATH_COUNT                      1501 non-null   int64 
 5   CASE_COUNT_7DAY_AVG              1501 non-null   int64 
 6   ALL_CASE_COUNT_7DAY_AVG          1501 non-null   int64 
 7   HOSP_COUNT_7DAY_AVG              1501 non-null   int64 
 8   DEATH_COUNT_7DAY_AVG             1501 non-null   int64 
 9   BX_CASE_COUNT                    1501 non-null   int64 
 10  BX_PROBABLE_CASE_COUNT           1501 non-null   int64 
 11  BX_HOSPITALIZED_COUNT            1501 non-null   int64 
 12  BX_DEATH_COUNT                   1

In [3]:
df.describe()

CASE_COUNT  PROBABLE_CASE_COUNT  HOSPITALIZED_COUNT  DEATH_COUNT  \
count   1501.000000          1501.000000         1501.000000  1501.000000   
mean    1940.461692           423.871419          140.890073    30.862758   
std     4371.937148           556.931279          219.265205    86.663209   
min        0.000000             0.000000            0.000000     0.000000   
25%      382.000000            95.000000           38.000000     4.000000   
50%      960.000000           301.000000           77.000000     9.000000   
75%     2157.000000           564.000000          145.000000    20.000000   
max    55058.000000          5882.000000         1858.000000   831.000000   

       CASE_COUNT_7DAY_AVG  ALL_CASE_COUNT_7DAY_AVG  HOSP_COUNT_7DAY_AVG  \
count          1501.000000              1501.000000          1501.000000   
mean           1940.241839              2364.009993           140.864091   
std            4029.178914              4510.710791           215.579969   
min               0.000000                 0.000000             0.000000   
25%             375.000000               511.000000            39.000000   
50%            1007.000000              1378.000000            76.000000   
75%            2311.000000              2874.000000           143.000000   
max           39531.000000             43988.000000          1687.000000   

       DEATH_COUNT_7DAY_AVG  BX_CASE_COUNT  BX_PROBABLE_CASE_COUNT  ...  \
count           1501.000000    1501.000000             1501.000000  ...   
mean              30.853431     321.758827               83.849434  ...   
std               85.634361     811.859029              126.838895  ...   
min                0.000000       0.000000                0.000000  ...   
25%                4.000000      59.000000               16.000000  ...   
50%               10.000000     139.000000               54.000000  ...   
75%               19.000000     329.000000              112.000000  ...   
max              775.000000   10565.000000             1575.000000  ...   

       SI_CASE_COUNT  SI_PROBABLE_CASE_COUNT  SI_HOSPITALIZED_COUNT  \
count    1501.000000             1501.000000            1501.000000   
mean      136.124584               30.612925               8.541639   
std       287.469866               33.261751              10.597630   
min         0.000000                0.000000               0.000000   
25%        26.000000                7.000000               2.000000   
50%        70.000000               22.000000               5.000000   
75%       156.000000               45.000000              11.000000   
max      3720.000000              316.000000              83.000000   

       SI_DEATH_COUNT  SI_PROBABLE_CASE_COUNT_7DAY_AVG  \
count     1501.000000                      1501.000000   
mean         1.881412                        30.598268   
std          4.051081                        30.933291   
min          0.000000                         0.000000   
25%          0.000000                         8.000000   
50%          1.000000                        25.000000   
75%          2.000000                        45.000000   
max         43.000000                       233.000000   

       SI_CASE_COUNT_7DAY_AVG  SI_ALL_CASE_COUNT_7DAY_AVG  \
count             1501.000000                 1501.000000   
mean               136.107928                  166.703531   
std                265.086887                  290.102055   
min                  0.000000                    0.000000   
25%                 27.000000                   36.000000   
50%                 75.000000                  103.000000   
75%                157.000000                  201.000000   
max               2689.000000                 2909.000000   

       SI_HOSPITALIZED_COUNT_7DAY_AVG  SI_DEATH_COUNT_7DAY_AVG    INCOMPLETE  
count                     1501.000000              1501.000000   1501.000000  
mean                         8.540306                 1.852099    284.616922  
std 

In [4]:
df.head()

date_of_interest  CASE_COUNT  PROBABLE_CASE_COUNT  HOSPITALIZED_COUNT  \
0       02/29/2020           1                    0                   1   
1       03/01/2020           0                    0                   1   
2       03/02/2020           0                    0                   2   
3       03/03/2020           1                    0                   7   
4       03/04/2020           5                    0                   2   

   DEATH_COUNT  CASE_COUNT_7DAY_AVG  ALL_CASE_COUNT_7DAY_AVG  \
0            0                    0                        0   
1            0                    0                        0   
2            0                    0                        0   
3            0                    0                        0   
4            0                    0                        0   

   HOSP_COUNT_7DAY_AVG  DEATH_COUNT_7DAY_AVG  BX_CASE_COUNT  ...  \
0                    0                     0              0  ...   
1                    0                     0              0  ...   
2                    0                     0              0  ...   
3                    0                     0              0  ...   
4                    0                     0              0  ...   

   SI_CASE_COUNT  SI_PROBABLE_CASE_COUNT  SI_HOSPITALIZED_COUNT  \
0              0                       0                      0   
1              0                       0                      0   
2              0                       0                      0   
3              0                       0                      0   
4              0                       0                      0   

   SI_DEATH_COUNT  SI_PROBABLE_CASE_COUNT_7DAY_AVG  SI_CASE_COUNT_7DAY_AVG  \
0               0                                0                       0   
1               0                                0                       0   
2               0                                0                       0   
3               0                                0                       0   
4               0                                0                       0   

   SI_ALL_CASE_COUNT_7DAY_AVG  SI_HOSPITALIZED_COUNT_7DAY_AVG  \
0                           0                               0   
1                           0                               0   
2                           0                               0   
3                           0                               0   
4                           0                               0   

   SI_DEATH_COUNT_7DAY_AVG  INCOMPLETE  
0                        0           0  
1                        0           0  
2                        0           0  
3                        0           0  
4                        0           0  

[5 rows x 55 columns]

In [5]:
df.tail()

date_of_interest  CASE_COUNT  PROBABLE_CASE_COUNT  HOSPITALIZED_COUNT  \
1496       04/04/2024         159                   80                  19   
1497       04/05/2024         136                   73                  16   
1498       04/06/2024          89                   59                   5   
1499       04/07/2024          82                   58                   0   
1500       04/08/2024          96                   33                   0   

      DEATH_COUNT  CASE_COUNT_7DAY_AVG  ALL_CASE_COUNT_7DAY_AVG  \
1496            0                  129                      200   
1497            0                  130                      202   
1498            0                  130                      202   
1499            0                  130                      201   
1500            0                  120                      184   

      HOSP_COUNT_7DAY_AVG  DEATH_COUNT_7DAY_AVG  BX_CASE_COUNT  ...  \
1496                   17                     1             26  ...   
1497                   17                     1             20  ...   
1498                   16                     0             14  ...   
1499                   14                     0             16  ...   
1500                   12                     0             17  ...   

      SI_CASE_COUNT  SI_PROBABLE_CASE_COUNT  SI_HOSPITALIZED_COUNT  \
1496             13                       3                      0   
1497             10                       9                      2   
1498             10                       6                      0   
1499              3                       5                      0   
1500             10                       7                      0   

      SI_DEATH_COUNT  SI_PROBABLE_CASE_COUNT_7DAY_AVG  SI_CASE_COUNT_7DAY_AVG  \
1496               0                                7                       9   
1497               0                                7                      10   
1498               0                                6                      11   
1499               0                                6                      10   
1500               0                                6                       9   

      SI_ALL_CASE_COUNT_7DAY_AVG  SI_HOSPITALIZED_COUNT_7DAY_AVG  \
1496                          16                               0   
1497                          17                               1   
1498                          17                               1   
1499                          16                               0   
1500                          16                               0   

      SI_DEATH_COUNT_7DAY_AVG  INCOMPLETE  
1496                        0       61030  
1497                        0       61030  
1498                        0       61030  
1499                        0       61030  
1500                        0       61030  

[5 rows x 55 columns]

I am not interested in all of the columns that are 7 day averages so the following cell will remove them.

In [6]:
# Define the columns to remove
columns_to_remove = list(range(5, 9)) + list(range(13, 18)) + list(range(22, 27)) + \
                    list(range(31, 36)) + list(range(40, 45)) + list(range(49, 55))

# Remove the specified columns
df.drop(df.columns[columns_to_remove], axis=1, inplace=True)

# Print the DataFrame to verify the columns are removed
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1501 entries, 0 to 1500
Data columns (total 25 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   date_of_interest        1501 non-null   object
 1   CASE_COUNT              1501 non-null   int64 
 2   PROBABLE_CASE_COUNT     1501 non-null   int64 
 3   HOSPITALIZED_COUNT      1501 non-null   int64 
 4   DEATH_COUNT             1501 non-null   int64 
 5   BX_CASE_COUNT           1501 non-null   int64 
 6   BX_PROBABLE_CASE_COUNT  1501 non-null   int64 
 7   BX_HOSPITALIZED_COUNT   1501 non-null   int64 
 8   BX_DEATH_COUNT          1501 non-null   int64 
 9   BK_CASE_COUNT           1501 non-null   int64 
 10  BK_PROBABLE_CASE_COUNT  1501 non-null   int64 
 11  BK_HOSPITALIZED_COUNT   1501 non-null   int64 
 12  BK_DEATH_COUNT          1501 non-null   int64 
 13  MN_CASE_COUNT           1501 non-null   int64 
 14  MN_PROBABLE_CASE_COUNT  1501 non-null   int64 
 15  MN_H

The following cell will remove the underscore character from the column names and replace with a space for better readability.

In [7]:
# Replace underscores with space strings in column names
df.columns = df.columns.str.replace('_', ' ')

# Print the updated DataFrame to verify the changes
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1501 entries, 0 to 1500
Data columns (total 25 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   date of interest        1501 non-null   object
 1   CASE COUNT              1501 non-null   int64 
 2   PROBABLE CASE COUNT     1501 non-null   int64 
 3   HOSPITALIZED COUNT      1501 non-null   int64 
 4   DEATH COUNT             1501 non-null   int64 
 5   BX CASE COUNT           1501 non-null   int64 
 6   BX PROBABLE CASE COUNT  1501 non-null   int64 
 7   BX HOSPITALIZED COUNT   1501 non-null   int64 
 8   BX DEATH COUNT          1501 non-null   int64 
 9   BK CASE COUNT           1501 non-null   int64 
 10  BK PROBABLE CASE COUNT  1501 non-null   int64 
 11  BK HOSPITALIZED COUNT   1501 non-null   int64 
 12  BK DEATH COUNT          1501 non-null   int64 
 13  MN CASE COUNT           1501 non-null   int64 
 14  MN PROBABLE CASE COUNT  1501 non-null   int64 
 15  MN H

Now to save the cleaned dataframe to a renamed csv file.

In [8]:
# Save the DataFrame as a CSV file
df.to_csv(r'C:\Users\Danat\OneDrive\Documents\CMP-262\FinalProject\Cleaned-NYC-COVID-19.csv', index=False)